components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__measurements.xml

```diff
@@ -108,51 +399,268 @@
+      #set ($sd = $mathtool.roundTo(2, $handler.valueToStandardDeviation($male, $ageMonths, $val)))
+      #set ($eval = "${percentile}#if($percentile&lt;=20 &amp;&amp; $percentile&gt;=4)th#elseif($percentile%10==1)st#elseif($percentile%10==2)nd#elseif($percentile%10==3)rd#{else}th#end pctl (#if ($sd &gt;= 0)+#end${sd}SD)")
     #end
-    #set ($sortedDataObjects = $sorttool.sort($sortedDataObjects, ['age:asc', 'date:asc']))
-    ##
-    #foreach ($obj in $sortedDataObjects )
-      #set ($o = $obj.obj)
-      #foreach($prop in $dataClass.properties)
-        #if ($prop.name == 'date' || $prop.name == 'age')
-          |= ${prop.prettyName}: |=(% colspan="2" %) $doc.display($prop.getName(), $o)
-        #elseif (!$prop.name.endsWith('_evaluation') &amp;&amp; "$!o.getProperty($prop.getName()).value" != '')
-          |= ${prop.prettyName}: | $doc.display($prop.getName(), $o) | $!doc.display("${prop.getName()}_evaluation", $o)
-        #end
+  #end
+  {{html clean="false" wiki="false"}}
+  &lt;label&gt;#if ("$!name" != "" &amp;&amp; $displayName)$services.localization.render("phenotips.UIXField.measurements.types.${name}"):#end&lt;/label&gt;
+  &lt;span class="side-label"&gt;#if ("$!side" != "")$services.localization.render("phenotips.UIXField.measurements.sides.${side}")#end#if ($handler.getComputationDependencies().size())&lt;span class="fa fa-bolt" title="$services.localization.render('phenotips.patientSheet.measurements.measurementIsComputedMsg')"&gt;&lt;/span&gt;#end&lt;/span&gt;
+  &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_type" value="${name}"$disabled&gt;
+  &lt;div class="metric"&gt;
+    #if ($xcontext.action == 'edit')
+      #if ($handler.getComputationDependencies().size())
+        &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_value" value="$!{val}"$disabled&gt;
+        &lt;span class="val"&gt;$!{val}&lt;/span&gt;
+      #else
+        &lt;input type="text" size="5" name="PhenoTips.MeasurementClass_${id}_value" value="$!{val}"$disabled&gt;
       #end
-    #end## foreach object
-    ##
-  #end## if dataclass
+      &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_age" value=""$disabled&gt;
+      &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_date" value=""$disabled&gt;
+      &lt;input type="hidden" name="PhenoTips.MeasurementClass_${id}_side" value="$!{side}"$disabled&gt;
+    #else
+      &lt;span class="val"&gt;$!{val}&lt;/span&gt;
+    #end
+    &lt;span class="unit"&gt;#if ("$!unit" != "")$services.localization.render("phenotips.UIXField.measurements.units.${unit}")#end&lt;/span&gt;
+  &lt;/div&gt;
+  &lt;span class="pctl#if ("$!val" != "") $services.measurements.getFuzzyValue($sd)#end"&gt;#if ("$!val" != "")$eval#end&lt;/span&gt;
+  {{/html}}
+)))
+#end##
+##
+## Source: http://extensions.xwiki.org/xwiki/bin/view/Extension/Sort+a+list+of+XWiki+objects
+#macro (sortXWikiObjectsList $inputObjects $propertyName $outputObjects)
```

Comment:
can we pull this out into some generic "util" macro, that's not specific to this particular sheet?